### PR TITLE
AWS Amplify App Module

### PR DESCRIPTION
**Pull request by sakruthijupalli**
This PR introduces a new Terraform module to provision and manage AWS Amplify apps. This module creates the necessary Amplify app configurations, branch deployments, and domain associations required for a fully functional Amplify app.

**Features**
- Configure an Amplify app with a GitHub repository
- Set up branch deployments for the Amplify app
- Configure domain association for the Amplify app

Usage
To use the module, add the following code block to your Terraform configuration:
```
module "aws_amplify" {
  source = "github.com/ManagedKube/kubernetes-ops//terraform-modules/aws/amplify?ref=add-amplify-app-module"

  name                  = "your-app-name"
  repository_url        = "https://github.com/your-org/your-repo"
  gh_access_token       = "your-github-access-token"
  branch_name           = "branch-to-deploy"
  domain_name           = "your-domain.com"
  sub_domain_prefix     = "subdomain-prefix"
  sub_domain_branch     = "subdomain-branch"
  build_spec            = "your-build-spec"
  environment_variables = {
    KEY1 = "value1"
    KEY2 = "value2"
  }
}
```
Screen Shot 2023-04-25 at 4 17 38 PM
Screen Shot 2023-04-25 at 4 31 10 PM
Screen Shot 2023-04-25 at 4 36 03 PM

### CODE REVIEW

File: terraform-modules/aws/amplify/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# AWS Amplify App Terraform Module
```
Comment (sekka1) [inline terraform-modules/aws/amplify/README.md:1]: `terraform-doc` should be ran on this module to create the docs.

Comment (sekka1) [reply terraform-modules/aws/amplify/README.md:1]: output will be like this: https://github.com/ManagedKube/kubernetes-ops/pull/428/files#diff-dfaa5dde6b8ced55169546ad5282f14812d3fcd7282c3ab2fd3a33cbb9af18e3R1

File: terraform-modules/aws/amplify/default_iam_policy.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
+{
```
Comment (sekka1) [inline terraform-modules/aws/amplify/default_iam_policy.json:1]: Can all of the 
```
            "Resource": "*"
```
be tightened up to be more specific?  Im pretty sure having a `*` here will sound off alarm bells on multiple IAM policy scanners that we have and are using.

File: terraform-modules/aws/amplify/main.tf [hunk 3]
```diff
@@ -0,0 +1,63 @@
+resource "aws_iam_role" "amplify" {
```
Comment (sekka1) [inline terraform-modules/aws/amplify/main.tf:1]: Should add in the `tags` key and pass in all of our standard tags

File: terraform-modules/aws/amplify/main.tf [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        Effect = "Allow"
+        Principal = {
+          Service = "amplify.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "amplify-policy" {
```
Comment (sekka1) [inline terraform-modules/aws/amplify/main.tf:18]: Tags

File: terraform-modules/aws/amplify/main.tf [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  policy = file("${path.module}/default_iam_policy.json")
+}
+# we have added this policy to resolve the AccessDenied Issue during the Code deployment
+# https://github.com/aws-amplify/amplify-hosting/blob/main/FAQ.md#error-accessdenied-access-denied
+resource "aws_iam_role_policy_attachment" "amplify-attach-policy" {
+  policy_arn = aws_iam_policy.amplify-policy.arn
+  role       = aws_iam_role.amplify.name
+}
+
+resource "aws_amplify_app" "amplify" {
```
Comment (sekka1) [inline terraform-modules/aws/amplify/main.tf:29]: tags

File: terraform-modules/aws/amplify/README.md (hunk 1 above)
Comment (sakruthijupalli) [reply terraform-modules/aws/amplify/README.md]: updated

File: terraform-modules/aws/amplify/default_iam_policy.json (hunk 2 above)
Comment (sakruthijupalli) [reply terraform-modules/aws/amplify/default_iam_policy.json:1]: updated

File: terraform-modules/aws/amplify/main.tf (hunk 3 above)
Comment (sakruthijupalli) [reply terraform-modules/aws/amplify/main.tf:1]: updated

File: terraform-modules/aws/amplify/default_iam_policy.json [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+              "s3:GetAccelerateConfiguration",
+              "s3:GetObject",
+              "s3:ListBucket",
+              "s3:PutAccelerateConfiguration",
+              "s3:PutBucketPolicy",
+              "s3:PutObject",
+              "s3:PutBucketTagging",
+              "s3:GetBucketTagging"
+            ],
+            "Resource": "arn:aws:s3:${aws_region}:${aws_account_id}:*"
```
Comment (sekka1) [inline terraform-modules/aws/amplify/default_iam_policy.json:83]: What buckets does this create?

File: terraform-modules/aws/amplify/default_iam_policy.json [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+              "sqs:CreateQueue",
+              "sqs:DeleteQueue",
+              "sqs:GetQueueAttributes",
+              "sqs:SetQueueAttributes"
+            ],
+            "Resource": "arn:aws:sqs:${aws_region}:${aws_account_id}:*"
```
Comment (sekka1) [inline terraform-modules/aws/amplify/default_iam_policy.json:93]: What sqs does this create?

File: terraform-modules/aws/amplify/main.tf [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        }
+      }
+    ]
+  })
+}
+
+#Base policy for Amplify app allows access to resources needed by Amplify applications.
+resource "aws_iam_role_policy_attachment" "role_attach" {
+  role       = aws_iam_role.amplify.name
+  policy_arn = "arn:aws:iam::aws:policy/AdministratorAccess-Amplify"
```
Comment (sekka1) [inline terraform-modules/aws/amplify/main.tf:21]: Document this as the source of this policy for future reference: https://docs.aws.amazon.com/amplify/latest/userguide/security-iam-awsmanpol.html?authuser=1

**Review by sekka1 — APPROVED** [review]
lgtm

**Pull request merged**